tests/drivers/pwm/pwm_api/drivers.pwm.pwm_fast fails on nrf54h20dk/nrf54h20/cpuapp

**Describe the bug**
The `tests/drivers/pwm/pwm_api/drivers.pwm.pwm_fast` test suite fails on `nrf54h20dk/nrf54h20/cpuapp` in CI on `main`:
https://github.com/zephyrproject-rtos/zephyr/actions/runs/11355771232/job/31594139033

This test suite was added in https://github.com/zephyrproject-rtos/zephyr/pull/75047

**To Reproduce**
Steps to reproduce the behavior:
1. `west twister -p nrf54h20dk/nrf54h20/cpuapp -s tests/drivers/pwm/pwm_api/drivers.pwm.pwm_fast`
2. See error below

**Expected behavior**
Test suite passes.

**Impact**
CI failure in `main`.

**Logs and console output**
```
FAILED: zephyr/zephyr_pre0.elf zephyr/zephyr_pre0.map /__w/zephyr/zephyr/twister-out/nrf54h20dk_nrf54h20_cpuapp/tests/drivers/pwm/pwm_api/drivers.pwm.pwm_fast/pwm_api/zephyr/zephyr_pre0.map 
: && ccache /opt/toolchains/zephyr-sdk-0.16.8/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc  -gdwarf-4 -gdwarf-4 zephyr/CMakeFiles/zephyr_pre0.dir/misc/empty_file.c.obj -o zephyr/zephyr_pre0.elf  zephyr/CMakeFiles/offsets.dir/./arch/arm/core/offsets/offsets.c.obj  -T  zephyr/linker_zephyr_pre0.cmd  -Wl,-Map=/__w/zephyr/zephyr/twister-out/nrf54h20dk_nrf54h20_cpuapp/tests/drivers/pwm/pwm_api/drivers.pwm.pwm_fast/pwm_api/zephyr/zephyr_pre0.map  -Wl,--whole-archive  app/libapp.a  zephyr/libzephyr.a  zephyr/arch/common/libarch__common.a  zephyr/arch/arch/arm/core/libarch__arm__core.a  zephyr/arch/arch/arm/core/cortex_m/libarch__arm__core__cortex_m.a  zephyr/arch/arch/arm/core/cortex_m/cmse/libarch__arm__core__cortex_m__cmse.a  zephyr/arch/arch/arm/core/mpu/libarch__arm__core__mpu.a  zephyr/lib/libc/picolibc/liblib__libc__picolibc.a  zephyr/lib/libc/common/liblib__libc__common.a  zephyr/soc/soc/nrf54h20/libsoc__nordic.a  zephyr/subsys/testsuite/ztest/libsubsys__testsuite__ztest.a  zephyr/drivers/cache/libdrivers__cache.a  zephyr/drivers/clock_control/libdrivers__clock_control.a  zephyr/drivers/console/libdrivers__console.a  zephyr/drivers/gpio/libdrivers__gpio.a  zephyr/drivers/pinctrl/libdrivers__pinctrl.a  zephyr/drivers/pwm/libdrivers__pwm.a  zephyr/drivers/serial/libdrivers__serial.a  zephyr/drivers/timer/libdrivers__timer.a  modules/hal_nordic/nrfx/libmodules__hal_nordic__nrfx.a  -Wl,--no-whole-archive  zephyr/kernel/libkernel.a  -L/__w/zephyr/zephyr/twister-out/nrf54h20dk_nrf54h20_cpuapp/tests/drivers/pwm/pwm_api/drivers.pwm.pwm_fast/pwm_api/zephyr  zephyr/arch/common/libisr_tables.a  -Wl,--fatal-warnings  -mcpu=cortex-m33  -mthumb  -mabi=aapcs  -mfp16-format=ieee  -mtp=soft  -fuse-ld=bfd  -Wl,--gc-sections  -Wl,--build-id=none  -Wl,--no-relax  -Wl,--sort-common=descending  -Wl,--sort-section=alignment  -Wl,-u,_OffsetAbsSyms  -Wl,-u,_ConfigAbsSyms  -nostdlib  -static  -Wl,-X  -Wl,-N  -Wl,--orphan-handling=warn  -Wl,-no-pie  -specs=picolibc.specs  -DPICOLIBC_LONG_LONG_PRINTF_SCANF -L"/opt/toolchains/zephyr-sdk-0.16.8/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/thumb/v8-m.main/nofp" -lc -lgcc && cd /__w/zephyr/zephyr/twister-out/nrf54h20dk_nrf54h20_cpuapp/tests/drivers/pwm/pwm_api/drivers.pwm.pwm_fast/pwm_api/zephyr && /usr/bin/cmake -E true
/opt/toolchains/zephyr-sdk-0.16.8/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/../../../../arm-zephyr-eabi/bin/ld.bfd: warning: orphan section `DMA_RAM21' from `zephyr/drivers/pwm/libdrivers__pwm.a(pwm_nrfx.c.obj)' being placed in section `DMA_RAM21'
collect2: error: ld returned 1 exit status

```

**Environment (please complete the following information):**
 - CI on `main`

--- a/tests/drivers/pwm/pwm_api/boards/nrf54h20dk_nrf54h20_cpuapp_fast.overlay
+++ b/tests/drivers/pwm/pwm_api/boards/nrf54h20dk_nrf54h20_cpuapp_fast.overlay
@@ -12,6 +12,10 @@
 	};
 };
 
+&dma_fast_region {
+	status = "okay";
+};
+
 &pwm120 {
 	status = "okay";
 	pinctrl-0 = <&pwm_default>;

